samza-test/src/test/scala/org/apache/samza/storage/kv/TestLargeMessageSafeKeyValueStores.scala

```diff
@@ -0,0 +1,184 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.storage.kv
+
+import java.io.File
+import java.util.{ArrayList, Arrays, Random}
+
+import org.apache.samza.config.MapConfig
+import org.apache.samza.serializers.StringSerde
+import org.apache.samza.storage.kv.TestLargeMessageSafeKeyValueStores.{BatchSize, CacheSize}
+import org.apache.samza.storage.kv.inmemory.InMemoryKeyValueStore
+import org.apache.samza.system.{OutgoingMessageEnvelope, SystemStreamPartition}
+import org.apache.samza.task.MessageCollector
+import org.apache.samza.{Partition, SamzaException}
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import org.junit.runners.Parameterized.Parameters
+import org.junit.{After, Before, Test}
+import org.scalatest.Assertions.fail
+
+@RunWith(value = classOf[Parameterized])
+class TestLargeMessageSafeKeyValueStores(typeOfStore: String, storeConfig: String, dropLargeMessageStr: String) {
```

Comment:
Can you please put a javadoc about what you are testing with this?

Reply:
Done.